app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoder.h

```diff
@@ -0,0 +1,150 @@
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedForMultiStreamWithPerfOld
+         */
+        Result DecodeInterleavedForMultiStreamWithPerfOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Same as DecodeInterleaved
+         * @see DecodeInterleaved
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleaved
+         */
+        Result DecodeInterleavedWithPerfAndResetOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Same as DecodeInterleavedForMultiStream
+         * @see DecodeInterleavedForMultiStream
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedForMultiStream
+         */
+        Result DecodeInterleavedForMultiStreamWithPerfAndResetOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Takes an input boolean flag, an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size, decoded sample count and decode time taken in microseconds
+         * @note The bool flag indicates whether or not a reset of the decoder context is being requested
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleaved
+         */
+        Result DecodeInterleaved(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Takes an input boolean flag, an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus multistream source data to the output buffer and returns decoded data size, decoded sample count and decode time taken in microseconds.
+         * The bool flag indicates whether or not a reset of the decoder context is being requested
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedForMultiStream
+         */
+        Result DecodeInterleavedForMultiStream(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+      SERVICE_DECL(
+          SFUNC(0x0, IHardwareOpusDecoder, DecodeInterleavedOld),
+          //SFUNC(0x1, IHardwareOpusDecoder, SetContext),
+          //SFUNC(0x2, IHardwareOpusDecoder, DecodeInterleavedForMultiStreamOld),
+          //SFUNC(0x3, IHardwareOpusDecoder, SetContextForMultiStream),
+          SFUNC(0x4, IHardwareOpusDecoder, DecodeInterleavedWithPerfOld),
+          //SFUNC(0x5, IHardwareOpusDecoder, DecodeInterleavedForMultiStreamWithPerfOld),
+          SFUNC(0x6, IHardwareOpusDecoder, DecodeInterleavedWithPerfAndResetOld),
+          //SFUNC(0x7, IHardwareOpusDecoder, DecodeInterleavedForMultiStreamWithPerfAndResetOld),
+          SFUNC(0x8, IHardwareOpusDecoder, DecodeInterleaved),
+          //SFUNC(0x9, IHardwareOpusDecoder, DecodeInterleavedForMultiStream)
```

Comment:
Same as other comment, we don't want unimplemented service functions being defined.